src/commands/kv/mod.rs

```diff
@@ -63,8 +64,14 @@ pub fn get_namespace_id(target: &Target, binding: &str) -> Result<String, failur
     )
 }
 
-fn api_client(user: &GlobalUser) -> HttpApiClient {
-    HttpApiClient::new(Credentials::from(user.to_owned()))
+fn api_client(user: &GlobalUser) -> Result<HttpApiClient, failure::Error> {
+    HttpApiClient::new(
+        Credentials::from(user.to_owned()),
+        HttpApiClientConfig {
+            // Use 3 minute timeout instead of default 30-second one.
+            http_timeout: Duration::from_secs(3 * 60),
```

Comment:
one thing to note: the KV API endpoints vary in _their_ configured timeout; we should make sure we know what it looks like when the KV api times out, vs when we time out, and how wrangler handles that.

Reply:
I looked into this. :) A timeout from the reqwest client looks like
```
Error: Error: https://api.cloudflare.com/client/v4/accounts/c2c4de4232cdae687dc7a4a0323df63a/storage/kv/namespaces: timed out
```
(^ this is what the original bug reporter observed).

A timeout from Cloudflare's end (e.g. the worker lags on fetching something from the backend) is represented by status code 504. I added new handling logic for this, so that Wrangler users are advised to wait a few seconds and re-run their command.